tests/client/api_restful/test_get_tips.c

```diff
@@ -17,35 +17,39 @@ void test_get_tips() {
 
   res_tips_t* res_tips = res_tips_new();
   TEST_ASSERT_NOT_NULL(res_tips);
+
   TEST_ASSERT(get_tips(&ctx, res_tips) == 0);
   if (res_tips->is_error == true) {
     printf("Error: %s\n", res_tips->u.error->msg);
   } else {
     TEST_ASSERT(get_tips_id_count(res_tips) > 0);
   }
+
   res_tips_free(res_tips);
 }
 
 void test_deser_get_tips() {
   char const* const json_tips =
-      "{\"data\":{\"tipMessageIds\":[\"0a7c22aa43620d938146bcfc94a40804b26a5aaf3913bdc51b2836c47650de5c\","
+      "{\"tipMessageIds\":[\"0a7c22aa43620d938146bcfc94a40804b26a5aaf3913bdc51b2836c47650de5c\","
       "\"3a3ca52a4c7d96fed4eef037db4421231f09e63a817bcf913f33c0806e565de9\","
       "\"a714ca72bb21d1e69b49a272713b9a1edc4d5679695680b5f36b907c0ed0d445\","
-      "\"ea878b98a3eb38154993ea27d597e6cbb8fda0cd7b71cc2eb345b6c62140a6bf\"]}}";
+      "\"ea878b98a3eb38154993ea27d597e6cbb8fda0cd7b71cc2eb345b6c62140a6bf\"]}";
 
   res_tips_t* res_tips = res_tips_new();
   TEST_ASSERT_NOT_NULL(res_tips);
-  TEST_ASSERT(deser_get_tips(json_tips, res_tips) == 0);
+
+  TEST_ASSERT(get_tips_deserialize(json_tips, res_tips) == 0);
   TEST_ASSERT(res_tips->is_error == false);
   TEST_ASSERT_EQUAL_INT(4, get_tips_id_count(res_tips));
   TEST_ASSERT_EQUAL_MEMORY("0a7c22aa43620d938146bcfc94a40804b26a5aaf3913bdc51b2836c47650de5c", get_tips_id(res_tips, 0),
```

Comment:
it only validates data without checking the null terminator. these are strings, we can use `TEST_ASSERT_EQUAL_STRING`.

Reply:
Fixed.